ods_ci/tests/Resources/Page/ODH/ODHDashboard/ODHDataScienceProject/Storages.resource

```diff
@@ -47,12 +47,39 @@ Storage Size Should Be
-    Run Keyword And Continue On Failure     Should Be Equal As Strings    ${oc_object_size}    ${size}Gi
-    ${rc}  ${oc_object_size}=    Run And Return Rc And Output   oc get pvc -n ${namespace} -o jsonpath='{.items[?(@.metadata.annotations.openshift\\.io/display-name=="${name}")].status.capacity.storage}'
+
+    ${rc}  ${pvc_status_phase}=    Run And Return Rc And Output
+    ...    oc get pvc -n ${namespace} -o jsonpath='{.items[?(@.metadata.annotations.openshift\\.io/display-name=="${name}")].status.phase}'    # robocop: disable
+    Should Be Equal As Integers    ${rc}    0    An error occurred during the check of PVC ${name} .status.phase value!
+    IF    "${pvc_status_phase}" == "Pending"
+        # PVC hasn't been bound yet - only simple text with max storage should be shown
+        ${storage_size_el}=    Set Variable
+        ...    ${STORAGE_SECTION_XP}//tr[@class="pf-c-table__expandable-row pf-m-expanded"]/td/div[strong[text()="Size"]]/div    # robocop: disable
+        Wait Until Page Contains Element    ${storage_size_el}    timeout=20s
+        ${displayed_size}=    Get Text    ${storage_size_el}
+        Run Keyword And Continue On Failure    Should Be Equal As Strings    ${displayed_size}    Max ${size}Gi
+    ELSE
+        # PVC is now or had been bound already sometime in past - there is:
+        #  - a number stating current usage
+        #  - a bar showing current usage of storage in a graphic manner
+        #  - and finally an acutal storage size number
+        # Note: it may take some time before UI updates the storage
+        #       info WRT cluster usage, look like it does so every 60s
+        ${bound_storage_size_el}=    Set Variable
+        ...    ${STORAGE_SECTION_XP}//tr[@class="pf-c-table__expandable-row pf-m-expanded"]/td/div[strong[text()="Size"]]/div/div[3]/div    # robocop: disable
+        Wait Until Page Contains Element    ${bound_storage_size_el}    timeout=65s
+        ${displayed_size}=    Get Text    ${bound_storage_size_el}
+        Run Keyword And Continue On Failure     Should Be Equal As Strings    ${displayed_size}    ${size}Gi
+        ${rc}  ${oc_object_size}=    Run And Return Rc And Output
+        ...    oc get pvc -n ${namespace} -o jsonpath='{.items[?(@.metadata.annotations.openshift\\.io/display-name=="${name}")].status.capacity.storage}'    # robocop: disable
+        Should Be Equal As Integers    ${rc}    0
+        ...    An error occurred during the check of PVC ${name} '.status.capacity.storage' value!
+        Run Keyword And Continue On Failure     Should Be Equal As Strings    ${oc_object_size}    ${size}Gi
+    END
+    ${rc}  ${oc_object_size}=    Run And Return Rc And Output
+    ...    oc get pvc -n ${namespace} -o jsonpath='{.items[?(@.metadata.annotations.openshift\\.io/display-name=="${name}")].spec.resources.requests.storage}'    # robocop: disable
+    Should Be Equal As Integers    ${rc}    0
```

Comment:
What is this check used for?

Reply:
Oh, okay - understand your question now!

So, this check was there even before - there were two `oc get pvc` checks before my changes, but this one is common for both "Pending" and "Bound" PVC statuses, whether the one that is in the ELSE branch now makes sense only for "Bound" case.

I'll add some comment there.